### PR TITLE
Consolidate brand identity resolution to prevent OTS branding leaks

### PR DESCRIPTION
**Pull request by delano**
## Summary

Fixes #3566

Centralizes brand identity resolution through `identityStore` to prevent "Onetime Secret" branding from leaking onto custom domains and unbranded installations. Previously, surfaces like MastHead, DefaultLogo, and page titles each re-derived the product-name fallback from raw bootstrap fields, creating multiple opportunities for the hardcoded OTS literal to appear where it shouldn't.

This change introduces:
- `identityStore.productName` — a neutral-safe product name resolver that replaces hand-rolled `brand_product_name ?? NEUTRAL_BRAND_DEFAULTS` snippets across surfaces
- `identityStore.showPlatformIdentity` — a base guard that suppresses the platform wordmark on custom domains (with or without uploaded logos) and whenever a per-tenant logo is present
- `resolveProductName()` helper in `constants/brand.ts` — shared by both component surfaces (via identityStore) and router guards (usePageTitle, which runs outside the i18n context)

## Key Changes

**identityStore.ts**
- Added `productName` computed property that applies the neutral-safe fallback chain
- Added `showPlatformIdentity` computed property that returns false for custom domains or when a per-tenant logo is present
- Both properties centralize decisions previously scattered across MastHead, DefaultLogo, and usePageTitle

**MastHead.vue**
- Now consumes `identity.productName` and `identity.showPlatformIdentity` instead of reading `brand_product_name` directly
- Removed hardcoded `NEUTRAL_BRAND_DEFAULTS` reference
- The `showPlatformIdentity` guard prevents the platform site name from appearing on custom domains

**DefaultLogo.vue**
- Now resolves the aria-label fallback through `identity.productName` instead of reading bootstrap fields directly
- Ensures the neutral default ("My App") is used when no brand name is configured, never "Onetime Secret"

**usePageTitle.ts**
- Removed hardcoded `APP_NAME = 'Onetime Secret'` constant
- Now uses `resolveProductName()` helper to apply the same neutral-safe fallback chain
- Ensures page titles, og:title, and twitter:title never leak OTS branding

**constants/brand.ts**
- Added `resolveProductName()` function as the single source of truth for the product-name fallback
- Used by both identityStore (component context) and usePageTitle (router-guard context)

## Test Plan

Comprehensive test coverage added:
- `identityStore.spec.ts`: Tests for `productName` and `showPlatformIdentity` resolution, including edge cases (empty string, null, undefined)
- `MastHead.customDomain.spec.ts`: Updated tests confirming the platform site name is suppressed on custom domains with no logo, while the neutral mark still renders
- `DefaultLogo.spec.ts`: Tests confirming the aria-label falls back to the neutral default via identityStore
- `usePageTitle.spec.ts`: Tests confirming page titles never emit OTS branding and properly fall back through the chain
- `brandingResolverConsolidation.guard.spec.ts`: Source-level guardrail tests ensuring surfaces don't re-derive brand identity or hardcode OTS literals

All existing tests pass; new tests verify the consolidation prevents the three leak classes (A1/A3/A4).

https://claude.ai/code/session_01MvR9x1zU2TBnmgQV43bD2p

### CODE REVIEW

File: src/shared/composables/usePageTitle.ts [hunk 1]
```diff
@@ -43,16 +43,33 @@ export function usePageTitle() {
...
+   * `identityStore.productName` uses — rather than the store itself, because
+   * usePageTitle runs in router guards (see setupRouterGuards), outside the
+   * component/i18n context that identityStore initializes with.
+   *
+   * Reads `.value` on each call so the resolved name stays reactive to store
+   * changes (do not hoist the raw values out of the refs).
    */
-  const getAppName = (): string => display_domain.value || APP_NAME;
+  const getAppName = (): string =>
+    display_domain.value || resolveProductName(brand_product_name?.value);
```
[the remark this answers is not in the document]
Comment (delano) [reply src/shared/composables/usePageTitle.ts:72]: This is intended — the `My App` fallback is the #3048/#3049 neutral-safe behavior, and the old `APP_NAME = 'Onetime Secret'` hardcode was precisely the A4 leak this PR removes.

To your conditional ("unless the backend now guarantees canonical installs populate `brand_product_name`"): the backend deliberately does **not** — `nil` is the intended default.

- `lib/onetime/models/custom_domain/brand_settings.rb:66` — `GLOBAL_DEFAULTS[:product_name] = nil` (its own comment points at the frontend `NEUTRAL_BRAND_DEFAULTS.product_name = 'My App'`).
- `apps/web/core/views/helpers/initialize_view_vars.rb:172` — `brand_product_name = brand_config['product_name'] || GLOBAL_DEFAULTS[:product_name]` → `nil` when unconfigured.
- `try/unit/web/page_title_brand_fallback_try.rb` has explicit regression guards that `brand_product_name`/`GLOBAL_DEFAULTS[:product_name]` are `nil` and that `page_title` must **never** equal `'Onetime Secret'`.

Reverting to preserve `'Onetime Secret'` would re-introduce the leak and contradict those guards. The canonical onetimesecret.com deployment shows its name because it sets `BRAND_PRODUCT_NAME`.

On **social meta**: `og:title`/`twitter:title` are server-rendered from the backend's `page_title` (which keeps a `|| site_name` tail fallback), and scrapers read the server HTML — so social previews are unaffected by this client-side change. The only surface that changes is the live browser tab, and only for a canonical install that sets neither `BRAND_PRODUCT_NAME` nor relies on the deprecated `site_name`. Frontend behavior is covered by `src/tests/shared/composables/usePageTitle.spec.ts` (unconfigured → `My App`, never `Onetime Secret`).

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/tests/shared/brandingResolverConsolidation.guard.spec.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  readFileSync(resolve(process.cwd(), relPath), 'utf-8');
+
+/**
+ * Strips comments so the "no re-derivation" assertions look only at executable
+ * code, not at documentation that legitimately names the very things we forbid
+ * in code (e.g. a comment explaining the A4 "Onetime Secret" leak).
+ */
+function stripComments(src: string): string {
+  return src
+    .replace(/<!--[\s\S]*?-->/g, '') // SFC HTML comments
```
[the remark this answers is not in the document]
Comment (delano) [reply src/tests/shared/brandingResolverConsolidation.guard.spec.ts]: Fixed in `0b53a19`. `stripComments` now re-applies the delimited-comment strip to a fixpoint (loops until the source stops changing), so a nested/adjacent pair can't leave a dangling `<!--` behind — the documented remediation for this pattern. Note this helper only processes trusted first-party source files (our own `.vue`/`.ts`) to gate the source-text guard assertions, so there was no injection surface; the fixpoint just keeps it robust and clears the alert. Added nested-comment invariant tests.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/tests/shared/brandingResolverConsolidation.guard.spec.ts [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+ * would miss (CodeQL js/incomplete-multi-character-sanitization). Line comments
+ * are stripped last and preserve `://` so URL literals survive.
+ */
+function stripComments(src: string): string {
+  let out = src;
+  let prev: string;
+  do {
+    prev = out;
+    out = out
+      .replace(/<!--[\s\S]*?-->/g, '') // SFC HTML comments
```
[the remark this answers is not in the document]
Comment (delano) [reply src/tests/shared/brandingResolverConsolidation.guard.spec.ts]: Re-fixed in `8a4d58a`. My first attempt (a fixpoint loop around the `.replace`) didn't clear this — the query flags the `String.replace(/<!--…-->/)` sanitizer call itself, regardless of any surrounding loop. So `stripComments` no longer uses `replace` at all: it's now a single left-to-right scan (`indexOf`/`slice`) that removes each comment region in one pass with no residue, leaving no sanitizer for the query to match. (As before: this helper only ever reads our own trusted source files, so there was never an injection surface — the rewrite just clears the finding and is strictly more correct.) Existing nested-comment invariant tests still pass.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by delano** [comment]
Thanks for the thorough review — closing out the five items:

1. **`stripComments` `//`-in-strings edge case** — addressed. The helper was rewritten (`8a4d58a`) as a single left-to-right scanner (no `String.replace`), which both clears the CodeQL sanitization finding and handles `://` correctly. A bare `//` inside a non-URL string literal is still treated as a comment, but the helper only ever reads our own trusted source files, so that limitation is harmless here.
2. **`showPlatformIdentity` suppressing the wordmark on canonical when a per-tenant logo exists** — intended and covered by the "false whenever a per-tenant logo is present, independent of strategy" test. The docstring already calls this out ("…and whenever a per-tenant logo is present"); canonical/subdomain are permitted *subject to* there being no per-tenant logo.
3. **`resolveProductName` is steps 2–3 only** — already documented in the `brand.ts` docstring ("Applies step 2 → step 3 of the fallback chain"); `usePageTitle` prepends step 1 (`display_domain`) itself. No change.
4. **`t('web.homepage.one_time_secret_literal', { product_name })` as a residual leak path** — good catch, and verified. In `en` (source) and 20 other locales the value is `{product_name}`, so it resolves to the neutral-safe `productName` (never leaks). But ~9 locales (ja, zh, fr_FR, fr_CA, de_AT, sv_SE, tr, pt_*, eo, mi_NZ) still carry a hardcoded translated brand for that key with a **stale `source_hash`** — they were translated against the old literal-brand en value and never re-synced. So `MastHead`'s logo `alt`/site-name does leak the platform brand in those locales on a private-label install. This is pre-existing translation debt rather than something this consolidation introduced, so we're **scoping it to a follow-up** rather than expanding this PR. (`OnetimeSecretLogo.vue`/`LegacyLogo.vue` also read that key, but those are intentionally the OTS-branded marks.)
5. **`storeToRefs` unused import in `identityStore.ts`** — it is used (the `storeToRefs(bootstrapStore)` destructure at the top of the store). No change.

---
_Generated by [Claude Code](https://claude.ai/code)_
